src/rustllvm/PassWrapper.cpp

```diff
@@ -969,11 +982,19 @@ LLVMRustCreateThinLTOData(LLVMRustThinLTOModule *modules,
   // linkage will stay as external, and internal will stay as internal.
   std::set<GlobalValue::GUID> ExportedGUIDs;
   for (auto &List : Ret->Index) {
+#if LLVM_VERSION_GE(5, 0)
+    for (auto &GVS: List.second.SummaryList) {
+#else
     for (auto &GVS: List.second) {
+#endif
       if (GlobalValue::isLocalLinkage(GVS->linkage()))
         continue;
       auto GUID = GVS->getOriginalName();
+#if LLVM_VERSION_GE(5, 0)
+      if (IsLiveByGUID(Ret->Index, GUID))
```

Comment:
I think this is just `GVS->flags().Live`?

Reply:
I copied from the relevant commit. I have no idea what's the pitfall.

Reply:
Let's switch to `GVS->flags().Live`.

Reply:
Please see [here](https://github.com/llvm-mirror/llvm/commit/ccb80b9c0f60f33780e5e29bf66a87bb56968b99#diff-594b163c3b86c2397deed7a1ea545e42R540).

Reply:
There's a few reasons we should not use this construction:

* The implementation of `IsLiveByGUID` goes from a GUID back to the `GlobalValueSummary` which is already in scope here as `GVS`.
* We don't need to take `WithGlobalValueDeadStripping` into account, this is all happening long before anything else.

Let's switch to `GVS->flags().Live`

Reply:
Fixed.